test_squash_mdblobs_multiple_chunks random failure on Linux

During a PR test the tests on Linux [reported the following](https://jenkins-ci.tribler.org/job/GH_Tribler_PR_Tests/job/PR_async_linux/691/):

```
  File "/usr/lib/python3.7/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/home/jenkins/.local/lib/python3.7/site-packages/asynctest/case.py", line 297, in run
    self._run_test_method(testMethod)
  File "/home/jenkins/.local/lib/python3.7/site-packages/asynctest/case.py", line 352, in _run_test_method
    result = method()
  File "/home/jenkins/ci/workspace/GH_Tribler_PR_Tests/PR_async_linux/tribler/src/tribler-core/tribler_core/tests/tools/test_as_server.py", line 53, in check
    result = fun(*argv, **kwargs)
  File "/home/jenkins/ci/workspace/GH_Tribler_PR_Tests/PR_async_linux/tribler/src/tribler-core/tribler_core/modules/metadata_store/tests/test_store.py", line 147, in test_squash_mdblobs_multiple_chunks
    for d in self.mds.process_compressed_mdblob(chunk2, skip_personal_metadata_payload=False)
  File "/usr/lib/python3.7/unittest/case.py", line 1045, in assertListEqual
    self.assertSequenceEqual(list1, list2, msg, seq_type=list)
  File "/usr/lib/python3.7/unittest/case.py", line 1027, in assertSequenceEqual
    self.fail(msg)
  File "/usr/lib/python3.7/unittest/case.py", line 680, in fail
    raise self.failureException(msg)
Lists differ: [b'z\[898 chars]bd\t', b'U\xfd\x08VZ\x15PH\xf3\x0b\xe8s\xed\x1[163 chars]x0c'] != [b'z\[898 chars]bd\t']

First list contains 1 additional elements.
First extra element 5:
b'U\xfd\x08VZ\x15PH\xf3\x0b\xe8s\xed\x1a\xd86=Yo\x1b\xc2\x8fh\xce\x10\x97F\xe1%\xd6u\xc4\x15\xe2\xe3f\x97\xa8\x0b\x12\xe3\xf5\x1b\n\x84\xbdR\xdd\xef\xa6\xbf\x83\x85\x16\xb4\xaa\x0f\xbc\xd3\x03\x90f\xc8\x0c'

  [b'z\x03\x1c\x9b\xd1\xcd,25Tu?u\x1f}O\xaciZ.T\x19U\xd6t\xbezI(\xbd/2'
   b'\xa4\xf0\xf7\x08Hjf\xddz\x8f\xf9M\xbb\xfd9\x10\xb5!v\xa3\x91.\x0bM'
   b'i\x8a\xea\xea:\xdfN\r',
   b'n\xc8\x85\x88\xa1\xe8\xa4\x99\xda/\xa6+=8\x89hdO7\xe1a~w\xe0\\L\xa1L'
   b'\x19E\xf8A\x99\xfcTy\xf1[\xe6-\xfag\xdb\x04 \xc4\xfc\x9f\xc7\xf3XgKQ\x01\xa4'
   b'\xb9\x9d\x8b~\rE\x05\x08',
   b'mfv\xd4\x1f8\x9a\x8fs\x99\x900Y.\x1b+\x96aFW\xd4u6\xa2;g\xfcU\xfd\xb9\xaeM'
   b'\xeaLzO\x15\x99\x0f\xf2\xec8\xca\xfbz\xa1\xc6W\x9a\xb8\x7f\xfd\x01\t\xd8w'
   b'v9P\xec\x9a\xfe\xc5\x00',
   b'\xba\xa3\xadt\x90\xe9\xff\xe7\x08##jU+\xe7\xcc\xe9\xefkZ\xd3\x14W\xab'
   b"\x17\xf0\xbc\x19\x96dZ\xd5\xf9;X.'\x8a\xa8\xc1\xab\xe7\x1a6S\xa9/\xf1"
   b'\xe5\xaa4\xc5\x8e\xe0\x8fOKg\xbc\xfe\x95m\xad\t',
   b'\xf8\xa0\xc0\xa4\xf4L\xe7\xdd\xfd\xec\xc4\xca\xe2a\xdd\xebLR(\xb1'
   b'\xc6\x91_\x91\x8f\x8b\xcf\x0c\xf9\xbf\xe5p\xcf\x8f\xefuY\x9f\xb1\xad'
-  b'\xf2\xf6W\xa1\xb0\xce8\xec|\xd5\xc4b!\x11\x06\xea\x15\x06x\x86\x91\x19\xbd\t',
?                                                                                 ^

+  b'\xf2\xf6W\xa1\xb0\xce8\xec|\xd5\xc4b!\x11\x06\xea\x15\x06x\x86\x91\x19\xbd\t']
?                                                                                 ^

-  b'U\xfd\x08VZ\x15PH\xf3\x0b\xe8s\xed\x1a\xd86=Yo\x1b\xc2\x8fh\xce\x10\x97F\xe1'
-  b'%\xd6u\xc4\x15\xe2\xe3f\x97\xa8\x0b\x12\xe3\xf5\x1b\n\x84\xbdR\xdd'
-  b'\xef\xa6\xbf\x83\x85\x16\xb4\xaa\x0f\xbc\xd3\x03\x90f\xc8\x0c']
-------------------- >> begin captured logging << --------------------
DEBUG   1592314254 selector_events:asyncio:53   Using selector: EpollSelector
DEBUG   1592314254           store:MetadataStore:344  ('Added payload batch to DB (entries, seconds): %i %f', (10, 0.008733))
DEBUG   1592314254           store:MetadataStore:344  ('Added payload batch to DB (entries, seconds): %i %f', (10, 0.007874))
--------------------- >> end captured logging << ---------------------
```

This was resolved after retesting.

It also seems to fail on Windows 64-bit (see [here](https://jenkins-ci.tribler.org/job/GH_Tribler_PR_Tests/job/PR_win64/886/artifact/output/test_runners_output/4.err/*view*/)):

```
======================================================================
FAIL: test_squash_mdblobs_multiple_chunks (test_store.TestMetadataStore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\tribler\appdata\local\programs\python\python38\lib\site-packages\asynctest\case.py", line 297, in run
    self._run_test_method(testMethod)
  File "c:\users\tribler\appdata\local\programs\python\python38\lib\site-packages\asynctest\case.py", line 352, in _run_test_method
    result = method()
  File "E:\jenkins\workspace\GH_Tribler_PR_Tests\PR_win64\tribler\src\tribler-core\tribler_core\tests\tools\test_as_server.py", line 53, in check
    result = fun(*argv, **kwargs)
  File "E:\jenkins\workspace\GH_Tribler_PR_Tests\PR_win64\tribler\src\tribler-core\tribler_core\modules\metadata_store\tests\test_store.py", line 143, in test_squash_mdblobs_multiple_chunks
    self.assertListEqual(
AssertionError: Lists differ: [b'\x[778 chars]\x02", b"\x08\xd1\r\xd6\xef\xdbg\xc5\x89\xdd8\[146 chars]b\t"] != [b'\x[778 chars]\x02"]

First list contains 1 additional elements.
First extra element 4:
b"\x08\xd1\r\xd6\xef\xdbg\xc5\x89\xdd8\xabO.\xcb<o\xe5:\xcd1'\xc9\x08\xf5\x88\xa0\xcf\x1d\xf1Q\x1fT\x85\x17\x97\x988\xc1\xe1D2;\x08]\xf5A\xaf\x1b\xa5$\x91;(\xb7km\xf4\x16\xd5\xbc\x9a\xdb\t"

  [b'\xcc\xc2\xc7EX\xc5\xa8\xe6"\x7f\xf3>\x96b\x02<\\\x80a\x19\xf6\xe8\xe4u'
   b'E\x06\xee\xf8\xa0\x82\xc8\xfc\xc0b\x0c|\xb9\x9fB\xd7\xe3\x06\x08{'
   b'\x03|\xca\x16\\\xc9\x06\x86\xfc\xb4@\xb0\x85\xfcg~y\xfa#\x00',
   b'PC\x07R\x15\x0b\xa5\xbeg\x90\x1e\x18\xe9v\x8e\xe7\xdd\xd9\x87q'
   b"\xf1\x02\xb6\xb5\x86\xd6r\x82\x85\xcc\xbc'\xf5\xc2\x8c\x8d\xc6Y\x82\x07"
   b"\x98\xb8\xe1$I\xda\xeft\xaaT'1\xd0\x8e\xdft]\x92\x8d\x1c\xc5\x05\xb0\x0e",
   b'\xf9\xde\xbca\n\xde\xd6O\xcc\x9b\x8d\x04\x80q\xc0W\xe0\x13\n@\xd0;u\xb9'
   b'\xef\x93\xce\x99/c\xd8\xdd\x07\x93\xdd\t\xea:\x118\xdbA\x91\xb0^CD\x7f'
   b"CAW\x1c\x12'\xa1\r,\xca\xa4\xa5\xa2xN\x04",
   b"\x96\x81\xe0\xc1F\xe7\x85m\xa0]\x82'\xbc\xfe+b?GNeH\xc0n\xce\x1eI\x8b\xe3"
   b'\xb7\x94\xa22#3C\xa4N\x16\x1f\xdc)\x80\x80\xe2\x9a\xe9~\xcb\xaa\xe5\xd2\xfa'
-  b'\x8f\xa9\xddx\x93/\x07\xb5M\xbc\xf2\x02',
?                                            ^

+  b'\x8f\xa9\xddx\x93/\x07\xb5M\xbc\xf2\x02']
?                                            ^

-  b"\x08\xd1\r\xd6\xef\xdbg\xc5\x89\xdd8\xabO.\xcb<o\xe5:\xcd1'\xc9\x08"
-  b'\xf5\x88\xa0\xcf\x1d\xf1Q\x1fT\x85\x17\x97\x988\xc1\xe1D2;\x08]\xf5A\xaf'
-  b'\x1b\xa5$\x91;(\xb7km\xf4\x16\xd5\xbc\x9a\xdb\t']
-------------------- >> begin captured logging << --------------------
DEBUG   1595331618 proactor_events:asyncio:623  Using proactor: IocpProactor
DEBUG   1595331618           store:MetadataStore:342  ('Added payload batch to DB (entries, seconds): %i %f', (10, 0.011719))
DEBUG   1595331618           store:MetadataStore:342  ('Added payload batch to DB (entries, seconds): %i %f', (10, 0.007813))
--------------------- >> end captured logging << ---------------------
```

--- a/src/tribler-core/tribler_core/modules/metadata_store/tests/test_channel_metadata.py
+++ b/src/tribler-core/tribler_core/modules/metadata_store/tests/test_channel_metadata.py
@@ -20,6 +20,7 @@ from tribler_core.modules.metadata_store.orm_bindings.channel_metadata import CH
 from tribler_core.modules.metadata_store.orm_bindings.channel_node import COMMITTED, NEW, TODELETE, UPDATED
 from tribler_core.modules.metadata_store.serialization import CHANNEL_TORRENT, COLLECTION_NODE, REGULAR_TORRENT
 from tribler_core.modules.metadata_store.store import MetadataStore
+from tribler_core.modules.metadata_store.tests.test_store import TEST_PERSONAL_KEY
 from tribler_core.tests.tools.base_test import TriblerCoreTest
 from tribler_core.tests.tools.common import TESTS_DATA_DIR, TORRENT_UBUNTU_FILE
 from tribler_core.utilities import path_util
@@ -34,8 +35,7 @@ class TestChannelMetadata(TriblerCoreTest):
     async def setUp(self):
         await super(TestChannelMetadata, self).setUp()
         self.torrent_template = {"title": "", "infohash": b"", "torrent_date": datetime(1970, 1, 1), "tags": "video"}
-        self.my_key = default_eccrypto.generate_key(u"curve25519")
-        self.mds = MetadataStore(":memory:", self.session_base_dir, self.my_key)
+        self.mds = MetadataStore(":memory:", self.session_base_dir, TEST_PERSONAL_KEY)
 
     async def tearDown(self):
         self.mds.shutdown()
@@ -106,14 +106,14 @@ class TestChannelMetadata(TriblerCoreTest):
         """
         Test whether the correct directory name is returned for channel metadata
         """
-        sample_channel_dict = TestChannelMetadata.get_sample_channel_dict(self.my_key)
+        sample_channel_dict = TestChannelMetadata.get_sample_channel_dict(TEST_PERSONAL_KEY)
         channel_metadata = self.mds.ChannelMetadata.from_dict(sample_channel_dict)
 
         self.assertEqual(len(channel_metadata.dirname), CHANNEL_DIR_NAME_LENGTH)
 
     @db_session
     def test_get_channel_with_dirname(self):
-        sample_channel_dict = TestChannelMetadata.get_sample_channel_dict(self.my_key)
+        sample_channel_dict = TestChannelMetadata.get_sample_channel_dict(TEST_PERSONAL_KEY)
         channel_metadata = self.mds.ChannelMetadata.from_dict(sample_channel_dict)
         dirname = channel_metadata.dirname
         channel_result = self.mds.ChannelMetadata.get_channel_with_dirname(dirname)

--- a/src/tribler-core/tribler_core/modules/metadata_store/tests/test_store.py
+++ b/src/tribler-core/tribler_core/modules/metadata_store/tests/test_store.py
@@ -2,12 +2,15 @@ import os
 import random
 import string
 from binascii import unhexlify
+from datetime import datetime
 
 from ipv8.database import database_blob
 from ipv8.keyvault.crypto import default_eccrypto
+from ipv8.keyvault.private.libnaclkey import LibNaCLSK
 
 from pony.orm import db_session, flush
 
+from tribler_core.modules.metadata_store.discrete_clock import clock
 from tribler_core.modules.metadata_store.orm_bindings.channel_metadata import CHANNEL_DIR_NAME_LENGTH, entries_to_chunk
 from tribler_core.modules.metadata_store.orm_bindings.channel_node import NEW
 from tribler_core.modules.metadata_store.serialization import (
@@ -29,6 +32,11 @@ from tribler_core.tests.tools.base_test import TriblerCoreTest
 from tribler_core.tests.tools.common import TESTS_DATA_DIR
 from tribler_core.utilities.path_util import str_path
 from tribler_core.utilities.random_utils import random_infohash
+
+TEST_PERSONAL_KEY = LibNaCLSK(
+    b'4c69624e61434c534b3af56022aa5d556c07aeed704ee98df7dca580f'
+    b'522e1405663f0d36508d2189cb8991af2dd27b34bc18b4d24869e2c4f2cfdb164a78ea6e687daf7a21640d62b1b'[10:]
+)
 
 
 def make_wrong_payload(filename):
@@ -57,8 +65,8 @@ class TestMetadataStore(TriblerCoreTest):
 
     async def setUp(self):
         await super(TestMetadataStore, self).setUp()
-        my_key = default_eccrypto.generate_key(u"curve25519")
-        self.mds = MetadataStore(":memory:", self.session_base_dir, my_key)
+        self.mds = MetadataStore(":memory:", self.session_base_dir, TEST_PERSONAL_KEY)
+        clock.clock = 1000
 
     async def tearDown(self):
         self.mds.shutdown()
@@ -103,6 +111,7 @@ class TestMetadataStore(TriblerCoreTest):
             self.mds.TorrentMetadata(
                 title=''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(20)),
                 infohash=database_blob(random_infohash()),
+                torrent_date=datetime.utcfromtimestamp(100),
             )
             for _ in range(0, 10)
         ]
@@ -124,6 +133,7 @@ class TestMetadataStore(TriblerCoreTest):
                 self.mds.TorrentMetadata(
                     title=''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(20)),
                     infohash=database_blob(random_infohash()),
+                    torrent_date=datetime.utcfromtimestamp(100),
                 )
                 for _ in range(0, 10)
             ]
@@ -131,6 +141,7 @@ class TestMetadataStore(TriblerCoreTest):
             chunk, index = entries_to_chunk(md_list, chunk_size=900)
             chunk2, _ = entries_to_chunk(md_list, chunk_size=900, start_index=index)
             dict_list = [d.to_dict()["signature"] for d in md_list]
+
             for d in md_list:
                 d.delete()
         self.assertListEqual(
@@ -140,6 +151,7 @@ class TestMetadataStore(TriblerCoreTest):
                 for d in self.mds.process_compressed_mdblob(chunk, skip_personal_metadata_payload=False)
             ],
         )
+
         self.assertListEqual(
             dict_list[index:],
             [

